static/app/components/organizations/environmentPageFilter/index.tsx

```diff
@@ -87,6 +91,10 @@ export function EnvironmentPageFilter({
 
   const handleChange = useCallback(
     async (newValue: string[]) => {
+      if (isEqual(newValue, envPageFilterValue)) {
+        return;
+      }
+
```

Comment:
If the new value is the same as the current value, don't do anything. Otherwise, clicking outside the menu will trigger a value commit and wipe out the desync state.